### PR TITLE
reproducible builds

### PR DESCRIPTION
**Pull request by karianna**
- First maven site stable cut of reproducible build
- Last fixes from full test run

### CODE REVIEW

**Comment by karianna** [comment]
When #66 then I'll rebase and the enforcer plugin will be happy :-)

File: pom.xml [hunk 1]
```diff
@@ -124,11 +171,56 @@
...
+                    <executions>
+                        <execution>
+                            <id>enforce-maven</id>
+                            <goals>
+                                <goal>enforce</goal>
+                            </goals>
+                            <configuration>
+                                <rules>
+                                    <requireMavenVersion>
+                                        <version>3.8.2</version>
```
Comment (dsgrieve) [inline pom.xml:197]: Everything else is a property. Why not this?

File: pom.xml [hunk 2]
```diff
@@ -237,17 +330,50 @@
...
         <plugins>
+            <plugin>
+                <groupId>com.github.spotbugs</groupId>
+                <artifactId>spotbugs-maven-plugin</artifactId>
+                <configuration>
+                    <plugins>
+                        <plugin>
+                            <groupId>org.mutabilitydetector</groupId>
+                            <artifactId>MutabilityDetector4FindBugs</artifactId>
+                            <version>0.9.1</version>
```
Comment (dsgrieve) [inline pom.xml:352]: Here too.

File: pom.xml [hunk 3]
```diff
@@ -259,10 +385,59 @@
...
+            </plugin>
         </plugins>
     </reporting>
+
     <profiles>
-        <!-- profile for downloading test log data,checks and running test-->
+        <!-- profile for downloading test log data,checks and running test -->
         <profile>
             <id>contributor</id>
             <properties>
```
Comment (dsgrieve) [inline pom.xml:443]: Side issue... not sure we need both skipTests and maven.test.skip.

File: pom.xml [hunk 4]
```diff
@@ -237,17 +330,50 @@
...
-                <groupId>com.github.spotbugs</groupId>
-                <artifactId>spotbugs-maven-plugin</artifactId>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
             </plugin>
 
         </plugins>
     </build>
 
+    <!-- If versions are not explicitly set it means they're inheriting from hte pluginManagement section above -->
```
Comment (dsgrieve) [inline pom.xml:341]: s/hte/the/

**Review by dsgrieve — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (karianna) [reply pom.xml:443]: I think one is picked up by plugins and the other captures the manual -DskipTests case.

**Pull request merged**